models/music_cont_out_proxy.cpp

```diff
@@ -0,0 +1,429 @@
+{
+}
+
+nest::music_cont_out_proxy::Buffers_::Buffers_()
+  : has_targets_( false )
+  , data_()
+{
+}
+
+nest::music_cont_out_proxy::Buffers_::Buffers_( const Buffers_& b )
+  : has_targets_( b.has_targets_ )
+  , data_( b.data_ )
+{
+}
+
+
+/* ----------------------------------------------------------------
+ * Parameter extraction and manipulation functions
+ * ---------------------------------------------------------------- */
+
+void
+nest::music_cont_out_proxy::Parameters_::get( DictionaryDatum& d ) const
+{
+  ( *d )[ names::port_name ] = port_name_;
+  ( *d )[ names::interval ] = interval_.get_ms();
+
+  ArrayDatum ad_record_from;
+
+  for ( size_t j = 0; j < record_from_.size(); ++j )
+  {
+    ad_record_from.push_back( LiteralDatum( record_from_[ j ] ) );
+  }
+
+  ( *d )[ names::record_from ] = ad_record_from;
+  std::vector< long_t >* pInd_map_long =
+    new std::vector< long_t >( target_gids_.size() );
+
+  std::copy< std::vector< long_t >::const_iterator,
+    std::vector< long_t >::iterator >(
+    target_gids_.begin(), target_gids_.end(), pInd_map_long->begin() );
+
+  ( *d )[ names::targets ] = IntVectorDatum( pInd_map_long );
+}
+
+void
+nest::music_cont_out_proxy::Parameters_::set( const DictionaryDatum& d,
+  const Node& self,
+  const State_& states,
+  const Buffers_& buffs )
+{
+
+  if ( states.published_ == false )
+  {
+    updateValue< string >( d, names::port_name, port_name_ );
+  }
+
+  if ( buffs.has_targets_
+    && ( d->known( names::interval ) || d->known( names::record_from ) ) )
+  {
+    throw BadProperty(
+      "The recording interval and the list of properties to record "
+      "cannot be changed after the index_map has been set." );
+  }
+
+
+  double_t v;
+  if ( updateValue< double_t >( d, names::interval, v ) )
+  {
+    if ( Time( Time::ms( v ) ) < Time::get_resolution() )
+    {
+      throw BadProperty(
+        "The sampling interval must be at least as long "
+        "as the simulation resolution." );
+    }
+
+    // see if we can represent interval as multiple of step
+    interval_ = Time::step( Time( Time::ms( v ) ).get_steps() );
+    if ( std::abs( 1 - interval_.get_ms() / v ) > 10
+        * std::numeric_limits< double >::epsilon() )
+    {
+      throw BadProperty(
+        "The sampling interval must be a multiple of "
+        "the simulation resolution" );
+    }
+  }
+  // extract data
+  if ( d->known( names::record_from ) )
+  {
+    record_from_.clear();
+
+    ArrayDatum ad = getValue< ArrayDatum >( d, names::record_from );
+    for ( Token* t = ad.begin(); t != ad.end(); ++t )
+    {
+      record_from_.push_back( Name( getValue< std::string >( *t ) ) );
+    }
+  }
+
+  if ( d->known( names::targets ) )
+  {
+    if ( record_from_.empty() )
+    {
+      throw BadProperty(
+        "The property record_from must be set before passing target_gids." );
+    }
+
+    if ( states.published_ == false )
+    {
+      ArrayDatum mca = getValue< ArrayDatum >( d, names::targets );
```

Comment:
I think this whole connection business needs to go to `calibrate()`. The way this is implemented currently, multiple calls to `SetStatus` are possible and would lead to multiple connections, which is not the desired behavior.